### DISCUSSION TITLE
Python newbie weird output

### DISCUSSION THREAD

**Discussion by Hikari334** [community/community#21448] [Other Feature Feedback, Questions, & Ideas]
<p>The code I posted below works accordingly to my expectations except for 1 problem. When it goes into the factorial route, after the number is determined as a prime number, ill see for example: 2! = 2none. The “none” here is the problem, but I have no idea where and what is causing that “none” to appear here. If there’s any suggestions or solutions, I would really appreciate it!</p>
<pre><code class="lang-auto">def getSum(number): 
    sum = 0
    for digit in str(number):  
      sum += int(digit)       
    return sum

def factorial():
    if(number&lt;0):
        print("invalid input")
    elif ((number==0) or (number==1)):
        print(number,end="! = 1")
    else:
        factorial=1
        for i in range(1, number+1):
            factorial=factorial*i
        print(number,end="! = "+ str(factorial))

number=int(input("enter number:")) #Q13
if(number==2):
    print("A prime number")
    print(factorial())
elif((number&lt;2) or (number%2==0)):
    print("not prime number")
    print(getSum(number))
else:
    for i in range (2, number):
        if (number%i==0):
            print("Not prime number")
            print(getSum(number))
            break
        if (i==number-1):
            print("A prime number")
            print(factorial())
        
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Hikari334" data-post="1" data-topic="187252">
<div class="title">
<div class="quote-controls"></div>
 Hikari334:</div>
<blockquote>
<pre><code class="lang-auto">    print(factorial())
</code></pre>
</blockquote>
</aside>
<p>This line is the reason: It writes the return value of the <code>factorial()</code> function to standard output. There is no explicit <code>return</code> statement in the function, so the return value is <code>None</code>.</p>
<p>There are a number of other issues with the <code>factorial()</code> function:</p>
<ul>
<li>
<p>It uses the global variable <code>number</code> instead of a parameter.</p>
</li>
<li>
<p>Don’t use the <code>end</code> parameter to <code>print()</code> for data. Instead better return the calculated value and format the result.</p>
</li>
<li>
<p>If the parameter <code>number</code> is invalid, the function should raise an error instead of just writing a message.</p>
</li>
</ul>

**Comment by Hikari334** [comment]
<p>Ah thanks man, got it fixed up!</p>
